### ISSUE TITLE
Problem with Quantization and Loss Function for Regression Tasks in Pre-training

### ISSUE THREAD

**Issue by whmoyou** [PriorLabs/TabPFN#744]
When handling regression tasks during the model pre-training stage, if continuous values are bucketed (quantized) into discrete classes and trained using cross-entropy loss, a fundamental issue arises: the cross-entropy loss cannot perceive the ordinal relationship or numerical distance between classes. For example, when the true class is 10, the loss incurred for predicting class 11 is the same as for predicting class 1. However, in actual regression tasks, the error of predicting 11 should be much smaller than predicting 1.

**Comment by iivalchev** [comment]
I have also observed instabilities and spiking negative loss when fine-tuning the regressor. However I am not sure that statement you make is entirely true. In the regressor the negative log _density_ of the true bucket is used which for high probability and narrow buckets could result in negative values.

**Comment by LennartPurucker** [comment]
Heyho, I would say the observation somewhat aligns with our internal findings.
We have been and are working on alternatives to avoid such inconsistencies going forward! 

@whmoyou thank you for raising the issue here. I will close this issue now, as it is not directly related to an issue in the code repository. Nevertheless, thank you for raising this! :)
